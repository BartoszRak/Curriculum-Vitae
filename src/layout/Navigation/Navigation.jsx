import React from 'react'
import PropTypes from 'prop-types'
import {
  withStyles, AppBar, Toolbar, Button,
} from '@material-ui/core'

import LanguageSelect from '~components/LanguageSelect'
import { useScrollTo } from '~hooks'

import style from './Navigation.style'

const routes = [
  {
    elementId: '#projects',
    name: 'Projects',
  },
  {
    elementId: '#bio',
    name: 'About me',
  },
]

export function Navigation({ classes }) {
  const { scrollTo } = useScrollTo()

  return (
    <AppBar
      classes={{
        root: classes.root,
      }}
      position="fixed"
    >
      <Toolbar className={classes.toolbar}>
        <div>
        {routes.map(route => (
          <Button
            classes={{
              label: classes.buttonLabel,
              root: classes.button,
            }}
            color="secondary"
            key={route.name}
            onClick={() => scrollTo(route.elementId)}
          >
            {route.name}
          </Button>
        ))}
        </div>
        <div>
          <LanguageSelect />
        </div>
      </Toolbar>
    </AppBar>
  )
}

Navigation.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Navigation)
