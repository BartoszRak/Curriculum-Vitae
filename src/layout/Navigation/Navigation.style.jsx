export default ({ spacing }) => ({
  button: {
    marginRight: spacing.unit * 4,
  },
  buttonLabel: {
    fontWeight: 'normal',
    fontSize: '1rem',
  },
  root: {
    boxShadow: 'none',
    padding: spacing.unit * 2,
    background: 'linear-gradient(180deg, rgba(0, 0, 0, 0.15) 0%, rgba(0, 0, 0, 0)100%)',
    margin: 0,
    padding: 0,
  }
})
