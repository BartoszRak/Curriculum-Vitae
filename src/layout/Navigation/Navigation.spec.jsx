import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { Navigation } from './Navigation'

jest.unmock('@material-ui/core')

jest.mock('./MobileMenu', () => 'MobileMenuMock')
jest.mock('./DesktopMenu', () => 'DesktopMenuMock')
jest.mock('~components/LanguageSelect', () => 'LanguageSelectMock')

let props

describe('component: Navigation', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      intl: {
        formatMessage: jest.fn(),
      },
    }
  })
  describe('rendering', () => {
    test('render without crush', () => {
      const wrapper = shallow(<Navigation {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = shallow(<Navigation {...props} />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render mobile and desktop menu', () => {
      const wrapper = shallow(<Navigation {...props} />)

      expect(wrapper.find('MobileMenuMock').length).toBe(1)
      expect(wrapper.find('DesktopMenuMock').length).toBe(1)
    })

    test('render language select', () => {
      const wrapper = shallow(<Navigation {...props} />)

      expect(wrapper.find('LanguageSelectMock').length).toBe(1)
    })
  })
})
