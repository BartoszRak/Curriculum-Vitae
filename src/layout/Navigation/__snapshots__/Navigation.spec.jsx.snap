// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Navigation should render as expected 1`] = `
<AppBarMock
  classes={
    Object {
      "root": undefined,
    }
  }
  position="fixed"
>
  <ToolbarMock>
    <ButtonMock
      classes={
        Object {
          "label": undefined,
          "root": undefined,
        }
      }
      key="Projects"
      onClick={[Function]}
    >
      Projects
    </ButtonMock>
    <ButtonMock
      classes={
        Object {
          "label": undefined,
          "root": undefined,
        }
      }
      key="About me"
      onClick={[Function]}
    >
      About me
    </ButtonMock>
  </ToolbarMock>
</AppBarMock>
`;
